homeassistant/components/climate/melcloud.py

```diff
@@ -0,0 +1,635 @@
+"""
+Support for MelCloud climates.
+
+Example configuration:
+
+climate:
+  - platform: melcloud
+    email: my@email.com
+    password: MYPASSWORD
+    lease_time: 60
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/melcloud/
+"""
+
+import logging
+import time
+import requests
+
+from homeassistant.components.climate import (
+    SUPPORT_FAN_MODE, SUPPORT_ON_OFF, SUPPORT_OPERATION_MODE,
+    SUPPORT_SWING_MODE, SUPPORT_TARGET_TEMPERATURE, ClimateDevice)
+from homeassistant.const import ATTR_TEMPERATURE, TEMP_CELSIUS
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'melcloud'
+REQUIREMENTS = []
+DEPENDENCIES = []
+
+SUPPORT_FLAGS = (SUPPORT_TARGET_TEMPERATURE |
+                 SUPPORT_FAN_MODE |
+                 SUPPORT_OPERATION_MODE |
+                 SUPPORT_ON_OFF |
+                 SUPPORT_SWING_MODE)
+
+
+OPERATION_HEAT_STR = "Heat"
+OPERATION_COOL_STR = "Cool"
+OPERATION_FAN_STR = "Fan"
+OPERATION_AUTO_STR = "Auto"
+OPERATION_OFF_STR = "Off"
+OPERATION_DRY_STR = "Dry"
+
+MELCLOUD_API_URL = "https://app.melcloud.com/Mitsubishi.Wifi.Client"
+
+
+class Language:
+    """List of language available."""
+
+    English = 0
+    German = 4
+    Spanish = 6
+    French = 7
+    Italian = 19
+
+
+class Mode:
+    """List of mode available."""
+
+    Heat = 1
+    Dry = 2
+    Cool = 3
+    Fan = 7
+    Auto = 8
+
+
+class MelCloudAuthentication:
```

Comment:
The device specific code must be moved to a module which is [published on PyPI](https://developers.home-assistant.io/docs/en/development_checklist.html).

Reply:
Regarding this point, Could you accept the component without this change ?
To be honest with you, I will not have enough time to change this.
I send you this PR  because lot of peoples requested it but I was not aware it was so restrictive.

If this component doesn't match your requierement I can understand it. Just reject it or let another contributor update it. Unfortunatly, I will not have enough time to do this change.

Thanks for your work on HA.
Best regards,
o0zz

Reply:
We can't accept the component without changing to use a standalone library for the device specific code.

I'll go ahead and close this PR. You can open a new PR if you have more time in the future.